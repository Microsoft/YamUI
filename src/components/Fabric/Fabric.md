### Notes for use

You should wrap your entire app in a `Fabric` component. The Fabric library relies on this component to manage some overall styling, a language-specific font stack, and some special styling for accessible keyboard tabbing. By wrapping your entire app you can ensure that all content conforms to the same baseline styles.

NOTE: Some codebases are not purely React and cannot be wrapped in a React component like that. We will need to contribute directly back to the Fabric library to extract the `Fabric` component's functionality into a utility module so it can also be used to manage CSS classes on a static wrapper DOM node.

Also note that any legacy (non-React) overlays in your app which render at the end of the body outside of your wrapping `Fabric` component, like a jQuery ToolTip or Lightbox, should wrap their own content in a `Fabric` component to ensure proper styling and typography. This is not necessary for any YamUI/Fabric popup-type components because they automatically nest their content in another `Fabric` component.

### Examples

Basic usage:

```js
<div>
  <Fabric>
    Entire application
  </Fabric>

  {/* Legacy lightbox rendered outside the main Fabric wrapper. */}
  <div className="lightbox">
    <Fabric>
      Lightbox content
    </Fabric>
  </div>
</div>
```
