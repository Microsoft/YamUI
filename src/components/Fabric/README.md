# Fabric

The `<Fabric />` component is provided directly by the Fabric library and manages typography and other styles for the entire nested app. It manages styles dynamically, changing some classes as elements receive and lose focus. Eventually, as Fabric continues to adopt Glamor, it will set language-specific fonts. All visible markup should be wrapped in a single `Fabric` component.

## Properties

(none)

## Notes for use

You should wrap your entire app in a `<Fabric>` component. The Fabric library relies on this component to manage some overall styling, a language-specific font stack, and some special styling for accessible keyboard tabbing. By wrapping your entire app you can ensure that all content conforms to the same baseline styles.

NOTE: Some codebases are not purely React and cannot be wrapped in a React component like that. We will need to contribute directly back to the Fabric library to extract the `<Fabric>` component's functionality into a utility module so it can also be used to manage CSS classes on a static wrapper DOM node.

Also note that any legacy (non-React) overlays in your app which render at the end of the body outside of your wrapping `<Fabric>` component, like a jQuery ToolTip or Lightbox, should wrap their own content in a `<Fabric>` component to ensure proper styling and typography. This is not necessary for any YamUI/Fabric popup-type components because they automatically nest their content in another`<Fabric>` component.

## Examples

### Basic usage

```html
<!doctype html>
<html dir="ltr">
  <body>
    <Fabric>
      Entire application
    </Fabric>

    <!-- Legacy lightbox etc. rendering at the end of the body outside our Fabric component would
         also need its own Fabric wrapper. -->
    <div id="lightbox">
      <Fabric>
        Lightbox content
      </Fabric>
    </div>
  </body>
</html>
```
