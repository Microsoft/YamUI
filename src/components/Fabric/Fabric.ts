/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import '../../yamui';
import { Fabric } from 'office-ui-fabric-react/lib/Fabric';

/**
 * The `Fabric` component is provided directly by the Fabric library and manages typography and
 * other styles for the entire nested app. It manages styles dynamically, changing some classes as
 * elements receive and lose focus. Eventually, as Fabric continues to adopt Glamor, it will set
 * language-specific fonts. All visible markup should be wrapped in a single `Fabric` component.
 */
export default Fabric;
