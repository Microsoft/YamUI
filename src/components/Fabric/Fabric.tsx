/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import '../../yamui';
import { Fabric } from 'office-ui-fabric-react/lib/Fabric';
export default Fabric;
