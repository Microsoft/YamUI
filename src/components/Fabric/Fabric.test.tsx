/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Fabric from '.';

describe('<Fabric />', () => {
  let component: ShallowWrapper;

  describe('with default options', () => {
    beforeEach(() => {
      component = shallow(<Fabric>wrapped content</Fabric>);
    });

    it('renders its given content', () => {
      expect(component.render().text()).toEqual('wrapped content');
    });

    it('contains its base <Fabric /> component class', () => {
      expect(component.render().find('.ms-Fabric').length).toEqual(1);
    });
  });
});
