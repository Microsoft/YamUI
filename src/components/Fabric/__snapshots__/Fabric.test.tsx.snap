// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Fabric /> with default options renders 1`] = `
<div
  class=
      ms-Fabric
      {
        color: #333333;
        font-size: 1.5rem;
        line-height: 2.0rem;
      }
      & button {
        font-family: inherit;
      }
      & input {
        font-family: inherit;
      }
      & textarea {
        font-family: inherit;
      }
      button {
        overflow: visible;
      }
>
  wrapped content
</div>
`;
